### PR TITLE
Improve login screen

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
This is a squashed and rebased version of PR #38506 

It might be easy to merge the code from here, rather than the other PR, which has 50 commits.

I left the changelog the same, so that it links to the original PR. That has all the discussion, manual test results, etc.

## How Has This Been Tested?
CI, and see posts in #38506 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: core/js/js.js [hunk 1]
```diff
@@ -1443,6 +1443,28 @@ function initCore() {
...
+	var color;
+
+	if ($('p.info').length > 0) {
+		color = $('p.info').css('color');
+	} else {
+		color = $('body').css('color');
+	}
+	
+	$('#body-login form label').css({
+		color: $('p.info').css('color')
```
Comment (JanAckermann) [inline core/js/js.js:1458]: Should we make use of the var color, here ? otherwise it wont get used

**Review by AlexAndBear — CHANGES_REQUESTED** [review]

**Comment by JanAckermann** [comment]
image

Setup looks quite broken

**Comment by phil-davis** [comment]
This is the same code as already approved in PR #38506 

@hurradieweltgehtunter can you look at the review comments please? Or whoever is managing these code changes.

**Comment by JanAckermann** [comment]
Fixed setup:

image

**Comment by JanAckermann** [comment]
Datadir is too wide :-1:

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiWebdavMove1-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiProvisioning-v1-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiTags-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareCreateSpecialToRoot2-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiAuthWebDav-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareReshareToShares3-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareReshareToShares1-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiWebdavMove2-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareOperationsToRoot-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiAuth-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareCreateSpecialToShares1-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiSharePublicLink2-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareManagementToRoot-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareManagementToShares-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiShareReshareToRoot1-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>apiWebdavLocks2-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Comment by JanAckermann** [comment]
What the hack is going on ?

**Comment by JanAckermann** [comment]
Don't merge yet, wip

**Comment by mmattel** [comment]
Just a reminder, docs relevant, pls file a docs issue when close to merge.
@EParzefall fyi

**Review by hurradieweltgehtunter — APPROVED** [review]

**Comment by JanAckermann** [comment]
Let me rebase, are we ready then?

**Comment by JanAckermann** [comment]
image

Maybe some margin between the Setup database label and buttons?

**Comment by JammingBen** [comment]
LGTM code-wise. I really like the new look 👍 

One small addition to @JanAckermann 's [last comment](https://github.com/owncloud/core/pull/38821#issuecomment-860578678): The db-buttons seem to be a little bit "off" horizontally compared to the input fields.

**Comment by hurradieweltgehtunter** [comment]
Fixed the margin, cursor and horinzontal alignment issues

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingPublic1-chrome-mariadb10.2-php7.4</strong> failed. The build is cancelled...</summary>

**Review by AlexAndBear — APPROVED** [review]

File: core/js/setup.js [hunk 2]
```diff
@@ -115,16 +119,16 @@ $(document).ready(function() {
...
+	// 	// otherwise it gets overwritten
+	// 	setTimeout(function(){
+	// 		// calculate negative left margin
+	// 		// half of the difference of width and default bix width of 300
+	// 		// add 10 to clear left side padding of button group
+	// 		var leftMargin = (($('#databaseBackend').width() - 300) / 2 + 10 ) * -1;
+
+	// 		$('#databaseBackend').css('margin-left', Math.floor(leftMargin) + 'px');
+	// 	}, 250);
+	// }
```
Comment (phil-davis) [inline core/js/setup.js:133]: Do we want to leave this commented-out code here?

Comment (hurradieweltgehtunter) [reply core/js/setup.js:133]: No, I removed that in one of the last commits. Why is this back again

**Comment by phil-davis** [comment]
@hurradieweltgehtunter is this done now? 
If so, then "somebody" (tm) should squash the commits. (I can do that when we are confident that the code is finished)

**Review by hurradieweltgehtunter — APPROVED** [review]

**Comment by hurradieweltgehtunter** [comment]
Looks good so far and rady to merge IMO

**Comment by mmattel** [comment]
It is really hard for the docs team to track changes where we please to file a docs issue when a PR gets close to merge (https://github.com/owncloud/core/pull/38821#issuecomment-859575260) - you would support a lot filing an issue on your own so that changes made finding its way into the documentation. Do not think that others will do it for you - just do it yourself, it is not a big effort but really helps 😢

**Comment by hurradieweltgehtunter** [comment]
@mmattel sorry if this lead to frustration. I have the update of the docs in my mind but wanted to wait until PR got merged to get the final design. I'm on it rn

**Comment by phil-davis** [comment]
Note: https://github.com/owncloud/password_policy/issues/338

password_policy CI started failing last night. I suspect that it is related to changes here, including changes that happen when resetting a password. I first need to see if it is just test code (either in password_policy or core) that needs adjusting, or if there is any real functional regression.

**Comment by hurradieweltgehtunter** [comment]
@phil-davis I might have the solution. just pushing a new PR

**Pull request merged**